frontend/src/App.tsx

```diff
@@ -50,11 +62,66 @@ import "assets/css/theme.scss"
 import "./App.scss"
 import "assets/css/header.scss"
 
-// We update the elements list in the state in batches. This is how often we apply those batches.
+interface Props {}
+
+interface UserSettingsProps {
+  wideMode: boolean
+  runOnSave: boolean
+}
+
+interface State {
+  connectionState: ConnectionState
+  elements: Elements
+  reportId: string
+  reportName: string | null
+  reportRunState: ReportRunState
+  showLoginBox: boolean
+  userSettings: UserSettingsProps
+  dialog?: DialogProps | null
+  sharingEnabled?: boolean
+}
+
+interface InitializeMsg {
+  environmentInfo: {
+    streamlitVersion: string
+    pythonVersion: string
+  }
+  userInfo: {
+    installationId: string
+    email: string
+  }
+  config: {
+    maxCachedMessageAge: number
+    gatherUsageStats: boolean
+    sharingEnabled: boolean
+  }
+  commandLine: string
+  sessionState: StateChangeProto
+}
+
+interface StateChangeProto {
+  reportIsRunning: boolean
+  runOnSave: boolean
+}
```

Comment:
Protos have typing information already. You don't need these interfaces. See `frontend/src/autogen/proto.d.ts`